test/parallel_api/dynamic_selection/test_internal_inline_scheduler.pass.cpp

```diff
@@ -7,210 +7,235 @@
-  fake_selection_handle_t h;
-
-  std::atomic<int> ecount = 0;
-
-  for (int i = 1; i <= N; ++i) {
-    auto w = s.submit(h,
-           [&](int q, int i) {
-             ecount += i;
-             return 0;
-           }, i
-    );
-    w.wait();
-    int count = ecount.load();
-    if (count != 1) {
-      std::cout << "ERROR: scheduler did not execute all tasks exactly once\n";
-      return 1;
+int
+test_submit_and_wait_on_sync()
+{
+    const int N = 100;
+    TestUtils::int_inline_backend_t s;
+    fake_selection_handle_t h;
+
+    std::atomic<int> ecount = 0;
+
+    for (int i = 1; i <= N; ++i)
+    {
+        auto w = s.submit(
+            h,
+            [&](int q, int i)
+            {
+                ecount += i;
+                return 0;
+            },
+            i);
+        w.wait();
+        int count = ecount.load();
+        EXPECT_EQ(i * (i + 1) / 2, count, "ERROR: scheduler did not execute all tasks exactly once\n");
     }
-  }
-  std::cout << "wait_on_sync single element: OK\n";
-  return 0;
+    std::cout << "wait_on_sync: OK\n";
+    return 0;
 }
 
-int test_submit_and_wait_on_sync_empty() {
-  const int N = 0;
-  TestUtils::int_inline_backend_t s;
-  fake_selection_handle_t h;
-
-  std::atomic<int> ecount = 0;
-
-  for (int i = 1; i <= N; ++i) {
-    auto w = s.submit(h,
-           [&](int q, int i) {
-             ecount += i;
-             return 0;
-           }, i
-    );
-    w.wait();
-    int count = ecount.load();
-    if (count != 0) {
-      std::cout << "ERROR: scheduler did not execute all tasks exactly once\n";
-      return 1;
+int
+test_submit_and_wait_on_sync_single_element()
+{
+    const int N = 1;
+    TestUtils::int_inline_backend_t s;
+    fake_selection_handle_t h;
+
+    std::atomic<int> ecount = 0;
+
+    for (int i = 1; i <= N; ++i)
+    {
+        auto w = s.submit(
+            h,
+            [&](int q, int i)
+            {
+                ecount += i;
+                return 0;
+            },
+            i);
+        w.wait();
+        int count = ecount.load();
+        EXPECT_EQ(1, count, "ERROR: scheduler did not execute all tasks exactly once\n");
     }
-  }
-  std::cout << "wait_on_sync empty list: OK\n";
-  return 0;
+    std::cout << "wait_on_sync single element: OK\n";
+    return 0;
 }
 
-int test_properties() {
-  std::vector<int> v = {1,2};
-  TestUtils::int_inline_backend_t s(v);
-  auto v2 = s.get_resources();
-  auto v2s = v2.size();
-
-  if (v2s != v.size()) {
-    std::cout << "ERROR: universe size inconsistent with queried universe\n";
-  }
-
-  for (int i = 0; i < v2s; ++i) {
-    if (v[i] != oneapi::dpl::experimental::unwrap(v2[i])) {
-      std::cout << "ERROR: reported universe and queried universe are not equal\n";
-      return 1;
+int
+test_submit_and_wait_on_sync_empty()
+{
+    const int N = 0;
+    TestUtils::int_inline_backend_t s;
+    fake_selection_handle_t h;
+
+    std::atomic<int> ecount = 0;
+
+    for (int i = 1; i <= N; ++i)
```

Comment:
Good point.
It's from the initial commit "Added experimental feature for dynamic device selection" :
https://github.com/oneapi-src/oneDPL/commit/ea466d230fc4f74ccd298f752a2201b08a0900dc

Reply:
Also may be interested the correctness of all another assignments like `const int N = ` in dynamic selection tests.
Because there are assignments to `0`, `1`, `100` and etc.

Reply:
This is also a test for an empty submit but the wait is inside the loop so that has to be fixed. Might also be better to just remove the loop.

Reply:
Fixed